R/weighted_quantile.R

```diff
@@ -0,0 +1,58 @@
+#' Function that accepts weights and performs an exact weighted quantile
+#' computation from a vector of samples
+#'
+#' @param x numeric vector from which to calculate (weighted) sample quantiles
+#' @param weights numeric vector of weights for each `x` value
+#' @param normalize boolean specifying whether to normalize the `weights`.
+#'   Defaults to TRUE
+#' @param probs numeric vector of (quantile) probabilities
+#' @noRd
+#'
+#' @return a vector of weighted quantile values for the requested quantile levels
+
+weighted_quantile <- function(x, weights, normalize = TRUE, probs) {
```

Comment:
I suggest that we provide more default arguments so that the function does something if the user gives a minimal call.  I believe that a default of `weights = NULL` may be needed to avoid errors in the `is.null(weights)` check below if the user doesn't provide any inputs.  And I suggested the same values for `probs` that were used in the Hmisc function, which also correspond to the quantiles that are returned by `summary` in base R by default.
```suggestion
weighted_quantile <- function(x, weights = NULL, normalize = TRUE, probs = c(0, .25, .5, .75, 1)) {
```